src/lib.rs

```diff
@@ -262,6 +262,25 @@ pub enum Token<'a> {
     },
 }
 
+impl<'a> Token<'a> {
+    /// Returns the [`StrSpan`] encompassing all of the token.
+    pub fn span<>(&self) -> StrSpan<'a> {
+        match self {
+            &Token::Declaration { span, ..} => span,
```

Comment:
`&` is not needed.

Reply:
I'm aware that `Token` is `Copy` but it is still 112 bytes in size, so much bigger than a pointer, hence why I used `&self`.

We could however do:

```rust
impl<'a> Token<'a> {
    /// Returns the [`StrSpan`] encompassing all of the token.
    pub fn span(&self) -> &StrSpan<'a> {
        match self {
            Token::Declaration { span, ..} => span,
            Token::ProcessingInstruction { span, ..} => span,
            Token::Comment { span, ..} => span,
            Token::DtdStart { span, ..} => span,
            Token::EmptyDtd { span, ..} => span,
            Token::EntityDeclaration { span, ..} => span,
            Token::DtdEnd { span, ..} => span,
            Token::ElementStart { span, ..} => span,
            Token::Attribute { span, ..} => span,
            Token::ElementEnd { span, ..} => span,
            Token::Text { text, ..} => text,
            Token::Cdata { span, ..} => span,
        }
    }
}
```

Would you prefer?

Reply:
I meant we don't need `&` before `Token`.

The code above is fine, but you should replace `&StrSpan<'a>` with just `StrSpan<'a>`

Reply:
Should be ok now :)